Fix empty forkid and parenthash in l2block read from datastream

KyrinCode opened the pull request:
Fixes #1740 

Original l2block in proto doesn't have forkid and parenthash. I believe this is a bug of implementation because the type `FullL2Block` read from datastream has these two fields and now they are both empty. After this fix, they will be written to datastream correctly, and ReadBatches in data_stream_server.go can read out all fields correctly.

V-Staykov committed:
feat: override version printer in the cli (#1384)

MorettiGeorgiev committed:
feat: default zkevm.witness flag to false

V-Staykov committed:
fix: parse rpcBatchNumber correctly by string (#1395)

MorettiGeorgiev committed:
Merge pull request #1394 from 0xPolygonHermez/default-zkevm-witness-full-to-false

feat: default zkevm.witness flag to false

hexoscott committed:
Regular info tree updates (#1399)

* starting work on info tree updates during execution

* info tree updater in sequencer loop

* logging latest index for info tree updates

revitteth committed:
tweak(ci): kurtosis upgrade prep (#1361)

jhkimqd committed:
test: add unit tests for zk/txpool/pool.go (#1396)

* test: testnoncefromaddress

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* test: new test testnoncefromaddress

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* test: fix insufficient funds to test address

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* chore: cleanup

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* test: add test for processRemoteTxs()

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* test: change expected nonce to match cdk erigon pool behaviour

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* fix: revert nonce calculation logic in queued pool

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* chore: lint

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* fix: use []string to test all possible combinations for policy output

Signed-off-by: Ji Hwan <jkim@polygon.technology>

---------

Signed-off-by: Ji Hwan <jkim@polygon.technology>

hexoscott committed:
agg layer network sync fixes (#1388)

handles pre etrog events properly and filters on rollup

Co-authored-by: Max Revitt <max@revitt.consulting>

jhkimqd committed:
fix: change return value of ListContentAtACL and write test for it (#1373)

* fix: change return value of ListContentAtACL and write test for it

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* chore: cleanup + fix test cases

Signed-off-by: Ji Hwan <jkim@polygon.technology>

---------

Signed-off-by: Ji Hwan <jkim@polygon.technology>

Stefan-Ethernal committed:
feat: Decode witness to SMT (#1363)

* feat: Decode witness to SMT

* chore: warning fixes and simplifications

* test: use require in witness unit tests

* feat: simplifications in SMT state reader

* fix: address comment

* test: use requires

* Allocate array in getValueInBytes

ARR552 committed:
Fix query body in JSONRPCCall (#1410)

hexoscott committed:
quiet logging for witness generation unwinds (#1407)

* quiet logging for witness generation unwinds

* fix - integration

---------

Co-authored-by: Max Revitt <max@revitt.consulting>

tclemos committed:
add unit test for stage l1_info_tree (#1408)

hexoscott committed:
limit modexp call to 8192 bit inputs (#1391)

* limit modexp call to 8192 bit inputs

* logic change to mod exp revert rules

* mod len 0 logic in modExp

* remove comment from modexp

* more mod exp zk tweaks

cffls committed:
Fix resequence script due to new cast version (#1414)

MorettiGeorgiev committed:
fix: healthcheck EthAPI interface (#1416)

* fix: healthcheck EthAPI interface

* fix: GetBlockByNumber pointer for fullTx param in rpcdaemon health test

hexoscott committed:
immediate batch seal flag for counter overflows (#1417)

# Conflicts:
#	eth/ethconfig/config_zkevm.go
#	turbo/cli/default_flags.go
#	turbo/cli/flags_zkevm.go
#	zk/debug_tools/test-contracts/package.json
#	zk/stages/stage_sequence_execute.go

cffls committed:
Prevent zero tracer from panic in edge cases (#1411)

Commit:
WIP: add unit test to stage l1_sequencer_sync

revitteth committed:
feat(fork): banana acc input hash (#1419)

revitteth committed:
tweak(zkevm_api): allow 0x00 accinputhash (#1421)

V-Staykov committed:
fix(stage_baches): rpc resequence stop stage on unwind (#1297)

* fix(stage_baches): rpc resequence stop stage on unwind

* fix: tests

* fix: datastream channel write blocking

* fix: datastream blocking test

* fix: add wait on the datastream connect loop

* fix: merge problems

* fix: blockhash comparison in stage batches processor

* fix: download entries till reaching the amount in header

* fix: add go sum package

* feat: internal reconnect on each method in datastream client

* fix: do not disconnect on stage batches end

* feat: add ctx close in datastream reconnections

* fix: send stop command after normal stop of reading

* feat: retry a fixed number of times in stage batches

* fix: return error on ctx done

* fix: reverse daastream server version

* feat: print ds found block

* feat: added more logs in stage batches

* fix: check for sync limit in stage batches

* fix: sync limit in stage  batches

* refactor: make unwind test erros a bit more readable

* refactor: make unwind tests erorrs more readable

* refactor(ds_client): wrap connection read and write to set timeout

* fix: add timeout to test clients

* fix: stage batches limit

* feat: up datastream server version

* fix: up datastream server version

* fix: go sum

* fix: add error handling for set timeouts in datastream client

* fix: handle zero checkTImeout value

* fix: remove flag setting for datastream timeout

* fix: ci config

* fix: resequence test timeout

* fix: remove timeout from pre-london ci config

* refactor: error handling

* fix: stop stage on unwind

* fix: missing id in client

* fix: tests

* fix: tests

* fix: finish processing blocks on last entry reached

* feat: send stop command at start of new cycle to not get timedout by server

* fix: remove accidental commit folder

* fix: remove unneeded commit

* fix: tests

* fix: remove unnneeded return

* fix: get correct parent block hash

* fix: read correct blockhash

* fix: unwind on ds block unwind

* refactor: error handling in datastream and stage batches

* fix: remove unneeded sleep

* fix: add a small sleep interval in the entry loop

* fix: stop streaming on querying new stuff from ds client

* fix: buffer clear before new reads

* fix: sleep more in resequence test

* fix: cast call

* fix: remove wrong flag on cast

* fix: cast json flags in test

* feat: added wait time for block to be available on sync node

* fix: resequence block check test

* Fix 'client already started' error on finding common ancestor

* Add timeout

---------

Co-authored-by: Scott Fairclough <70711990+hexoscott@users.noreply.github.com>
Co-authored-by: Jerry <jerrycgh@gmail.com>

rachit77 committed:
Remove SMT logic for Zero Prover (#1374)

* wip

* wip

* apply feedback

* Fix nil state root when there isn't changes in-between blocks

* apply feedback

---------

Co-authored-by: Jerry <jerrycgh@gmail.com>

revitteth committed:
fix(zkevm_api): post etrog virtual batch seq no 0 (#1426)

hexoscott committed:
set a minimum stream client timeout (#1427)

hexoscott committed:
only discard a transaction if the batch is completely empty (#1431)

blocks take up counters so this tx might be valid at the start of the very next batch

hexoscott committed:
always use most recent info tree index for new blocks (#1429)

cffls committed:
Allow execution to short circuit to the last downloaded batch (#1432)

Commit:
implement test cases for TestSpawnL1SequencerSyncStage

Commit:
add more test cases for TestSpawnL1SequencerSyncStage

Commit:
Merge branch 'zkevm' into test/stage-l1-sequencer-sync

tclemos committed:
Merge pull request #1435 from 0xPolygonHermez/test/stage-l1-sequencer-sync

test: add unit test for Stage: l1_sequencer_sync

revitteth committed:
fix(zkevm_api): feed accinputhash back to accinputhash calc (#1438)

revitteth committed:
tweak(zkevm_api): check if sequencer for closed batch (#1439)

revitteth committed:
tweak(zkevm_api): improve accinput missing batch log (#1441)

hexoscott committed:
fall back method for checking batch closed state (#1442)

V-Staykov committed:
fix: setting correct timeout on buffer emptying (#1437)

* fix: setting correct timeout on buffer emptying

* fix: handle sent stop command error

* fix: send stop command if already started error occurs

* fix: nullate errors on loops

* fix: add stops after all start commands

* fix datastream unit test after new timeout code is in place

* fix for test DS client

* DS test timeout updates

* test helper for stream client

* change to handling isStreaming logic in DS client

* missing function on DS test client

---------

Co-authored-by: Scott Fairclough <scott@hexosoft.co.uk>

V-Staykov committed:
fix: use stage_execute block numbers for a sequencer in rpc calls (#1444)

hexoscott committed:
increase timeouts for unwind tests (#1445)

* increase timeouts for unwind tests

* remove timeout on unwind stream host

MorettiGeorgiev committed:
feat: add mock witness generation (#1436)

* feat: add mock witness generation base on flag and executorUrls emptiness

* chore: remove logging of index for mock witness generation

* feat: log start and end block number, instead of each block for mock witness

revitteth committed:
tweak(zkevm_api): accinput batch 0/1 0x00...0 (#1446)

jhkimqd committed:
feat: rollupaddress and rollupmanageraddress rpc method

Signed-off-by: Ji Hwan <jkim@polygon.technology>

jhkimqd committed:
test: add unit tests for zkevm_getrollupaddress and zkevm_getrollupmanageraddress

Signed-off-by: Ji Hwan <jkim@polygon.technology>

jhkimqd committed:
chore: cleanup

Signed-off-by: Ji Hwan <jkim@polygon.technology>

jhkimqd committed:
chore: lint

Signed-off-by: Ji Hwan <jkim@polygon.technology>

V-Staykov committed:
fix: health check block timestamp fix (#1448)

jhkimqd committed:
Merge pull request #1447 from jhkimqd/jihwan/new-zkevm-api

feat: new zkevm api to retrieve rollup address and rollup manager address

cffls committed:
Add CPU monitor in CI (#1389)

* Add CPU monitor in CI

* feat: adding make command to run it also.

* feat: fixing script to also run on macOS, adding README to future newcomers

---------

Co-authored-by: Arthur Abeilice <afa7789@gmail.com>

hexoscott committed:
bug fix in close batch status for RPC node (#1449)

Commit:
WIP: data stream refactoring to testable code; add data stream catch up test; pending asserts

jhkimqd committed:
test: unit tests for zk/txpool/send.go (#1420)

* test: add initial unit tests for zk/txpool/send.go

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* chore: temporarily remove failing tests

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* test: use functions under zk directory

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* fix: fix failing unit tests

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* chore: cleanup tests

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* chore: lint

Signed-off-by: Ji Hwan <jkim@polygon.technology>

---------

Signed-off-by: Ji Hwan <jkim@polygon.technology>
Co-authored-by: Valentin Staykov <79150443+V-Staykov@users.noreply.github.com>

cffls committed:
Remove redundant writes when a state object is reverted (#21) (#1450)

* Remove redundant writes when a state object is reverted (#21)

* Remove redundant writes when a state object is reverted

* Change IsDirty to Transaction level

We don't want a reverted transaction to show up in written trace because it was touched by a previous transaction.

* Add storage read whenever there is a sstore

This fixes an issue when a storage slot is
* written but got reverted
* never read by sLoad opcode

When this happens, we still need to include the storage slot in the trace.

* fix test

jhkimqd committed:
test: fix tests for zkevm rollup address + rollup manager address calls (#1452)

Signed-off-by: Ji Hwan <jkim@polygon.technology>
Co-authored-by: Valentin Staykov <79150443+V-Staykov@users.noreply.github.com>

V-Staykov committed:
fix: defer rollback on new tx (#1453)

Commit:
add assertion to data stream catch up test

jhkimqd committed:
test: add test file for fetch.go  (#1401)

* test: add test file for fetch.go

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* chore: lint

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* chore: cleanup

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* docs: add docs in test code

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* test: generate mock for pool interface

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* test: add test for TxPool's OnNewBlock function

Signed-off-by: Ji Hwan <jkim@polygon.technology>

---------

Signed-off-by: Ji Hwan <jkim@polygon.technology>
Co-authored-by: Valentin Staykov <79150443+V-Staykov@users.noreply.github.com>

jhkimqd committed:
fix: follow mock generate convention being used in makefile to avoid pool_mock.go from being deleted

Signed-off-by: Ji Hwan <jkim@polygon.technology>

jhkimqd committed:
Merge pull request #1462 from jhkimqd/jihwan/fix-pool-mock-gen

fix: follow mock generate convention

jhkimqd committed:
test: add unit test for p256verifyzkevm

Signed-off-by: Ji Hwan <jkim@polygon.technology>

jhkimqd committed:
test: add test for bigmodexpzkevm

Signed-off-by: Ji Hwan <jkim@polygon.technology>

xavier-romero committed:
Fix modexp issue returning wrong result (#1468)

hexoscott committed:
witness forced storage inclusion (#1465)

hexoscott committed:
fix for deadlock in db concurrency (#1472)

revitteth committed:
fix(seq): latest block from execution for rpc (#1473)

tclemos committed:
Merge pull request #1456 from 0xPolygonHermez/test/data-stream-catch-up

test: add unit test for Stage: data_stream_catch_up

jhkimqd committed:
Merge branch 'zkevm' into jihwan/p256verify-test

jhkimqd committed:
fix: mod value not being padded + new test cases

Signed-off-by: Ji Hwan <jkim@polygon.technology>

hexoscott committed:
Fix/txpool pending (#1477)

* no overflow when configured and mining logic change

* remove mined transactions from the pool per block

Commit:
Fix modexp when partial base/exp/mod is set

V-Staykov committed:
refactor: some better error description and handling (#1454)

* refactor: some better error description and handling

* refactor: stage batches errors tracing

* refactor: l1syncer error trace

* refactor: l1syncer error traces

* fix: move execute block err in the correct place

jhkimqd committed:
Merge branch 'zkevm' into jihwan/p256verify-test

Signed-off-by: Ji Hwan <jkim@polygon.technology>

revitteth committed:
fix(syncer-utils): elderberry validium decoder (#1470)

revitteth committed:
tweak(txpool): trace logs for best (#1466)

revitteth committed:
tweak(zkevm_api): accinput batch 0/1 (#1464)

* tweak(zkevm_api): accinput batch 0/1

* tweak(zkevm_api): accinput batch 0/1

jhkimqd committed:
Merge pull request #1463 from jhkimqd/jihwan/p256verify-test

test: add unit test for p256verifyzkevm

laisolizq committed:
Fix modexp check bytes length

hexoscott committed:
preload sender and check for errors (#1480)

hexoscott committed:
discard transactions from the pool that error during execution (#1490)

hexoscott committed:
adding more unit tests around modexp

laisolizq committed:
Merge pull request #1489 from 0xPolygonHermez/fix/modexp-error-8192

Fix modexp check bytes length

revitteth committed:
fix(ds): client - defer set streaming (#1492)

revitteth committed:
AccInputHash - L1 Empty (#1455)

* feat(zkevm_api): accinputhash node local calculation

* fix(zkevm_api): tests

* fix(zkevm_api): tests

revitteth committed:
Unwind Test (#1500)

* Fix/2.60 rpc sync (#1496)

* simplify datastream error handling logic in stage batches

# Conflicts:
#	zk/datastream/client/stream_client.go

* interface missing on test stream client
# Conflicts:
#	zk/datastream/client/stream_client.go

* fix for nonce issue in RPC (#1501)

* add zkevm_getLatestDataStreamBlock rpc endpoint

# Conflicts:
#	turbo/jsonrpc/zkevm_api.go

* use rpc in place of stream for fork 12 unwind process

# Conflicts:
#	zk/stages/stage_batches.go

* making common ancestor search testable

* fix(.gitignore): changing gitignore to debug

* chore(unwind-test): small changes to be able to fix this thing here.

* chore(unwind-tests): changing script to run locally

* fix: mod value not being padded + new test cases

Signed-off-by: Ji Hwan <jkim@polygon.technology>

* chore(unwind-test): changing timeout times to try and make it run without failure

* chore(unwind-tests): adding a debugging log a change to the script and a test fix

* chore(unwind-tests): fixing file that went wrong on rebase.

* chore(unwind-tests): undoing change on test

* chore(unwind-tests): comment on a part of the code + changes on unwind to test on PR

* removing comments on action

* tweak(makefile): add test-unwind to makefile

* test(unwind): unwind test optimisations

* fix(kurtosis): batch monitor specify enclave (env var)

* fix(kurtosis): batch monitor specify enclave (env var)

* Revert "AccInputHash - L1 Empty (#1455)"

This reverts commit 8ee70a34cb945fd6b6b3d6094615d57cd3e72746.

* fix(docs): doc gen/check

---------

Signed-off-by: Ji Hwan <jkim@polygon.technology>
Co-authored-by: Scott Fairclough <70711990+hexoscott@users.noreply.github.com>
Co-authored-by: Scott Fairclough <scott@hexosoft.co.uk>
Co-authored-by: Arthur Abeilice <afa7789@gmail.com>
Co-authored-by: Ji Hwan <jkim@polygon.technology>

afa7789 committed:
feat(txpool): log nonce issues during transaction addition

hexoscott committed:
mined transactions removed from inclusion list (#1504)

same approach we use for removing bad transactions from this same list

hexoscott committed:
check for contract code ending in an empty push pre fork 10 (#1159)

* check for contract code ending in an empty push pre fork 10

* fix: wrong case usage

* fixing panic on len == 0, going to -1

* remove hard coded 0x60 for push1 check

---------

Co-authored-by: Arthur Abeilice <afa7789@gmail.com>

hexoscott committed:
purge routine for txpool (#1471)

tclemos committed:
test: add unit test for Stage: L1 syncer (#1474)

afa7789 committed:
feat(txpool): changes requested on PR

afa7789 committed:
feat(txpool): enhance logging for transaction replacement attempts

afa7789 committed:
Merge pull request #1507 from 0xPolygonHermez/1430-sequencer-debug-level-log-when-a-transaction-is-replaced

feat(txpool): log nonce issues during transaction addition

laisolizq committed:
Fix modexp checks (#1494)

* Fix modexp checks

* remove small comment for modexp

* update modexp tests

---------

Co-authored-by: Scott Fairclough <70711990+hexoscott@users.noreply.github.com>
Co-authored-by: Scott Fairclough <scott@hexosoft.co.uk>

V-Staykov committed:
Refactor(smt):  code quality (#1509)

* fix(smt): handle errors in updateDepth and SetDepth methods

* refactor(smt_batch): separate functions and repeating code

* fix: progress chan size

* refactor: function params new lines

* refactor: function params on new lines

V-Staykov committed:
feat: add witness cache stage (#1469)

* feat: add witness cache stage

* fix: merge problems

* fix: merge problems

* fix: tests

* fix: unwinds

* refactor: remove unused function

* fix: add witness cache table in erigon-lib

* feat: add merge witnesses and tests

* feat: batch witness get from db

* fix: minor refactoring and flag fixes

* fix: unwindzksmt argument

* refactor: prints and add test

* fix: tests

* fix: merge build

* fix: add tests and fix witness parsing bugs

* feat: implement witness utility function and improve error handling

* feat: add SMT node retrieval and utility functions

* feat: restore L1 Info tree stage functionality with sync, unwind, and prune methods

* fix: correct db table for delete caches

* fix: do not unwind if stage witness flag not set

* fix: quit channel in unwindsmt

* refactor: separate changes getter in a file

V-Staykov committed:
feat: add kurtosis setup composite action (#1531)

* feat: add kurtosis setup composite action

* feat: update queries and replace kurtosis setup wth composite

* fix: move enclave name

* fix: unify kurtosis setup action

* fix: add shell bash to action

* fix: export l2rpc endpint

V-Staykov committed:
feat(rpc): use old l1 gas price if l1 not available (#1524)

hexoscott committed:
sequencer to report correct block height (#1527)

MorettiGeorgiev committed:
fix: replace L1 rpc url in error for getting l1 gas price (#1533)

hexoscott committed:
do not build empty blocks on detecting an overflow (#1525)

also stop processing transactions when the block timer has ticked to
give more consistent block times

Stefan-Ethernal committed:
Encode transactions to RLP format in zero tracer (#1360)

* feat: set NewTxnTrieNode to TxnMeta and add some TODOs

* feat: remove NewTxnTrieNode and BlockUsedCodeHashes

* chore: minor simplifications

hexoscott committed:
handle nonce too high occurrences during sequencing (#1535)

also update the pool state for the sender as we update it with mined
or deleted transactions

revitteth committed:
CI Performance (#1506)

* perf(ci): kurtosis runs parallel

* perf(ci): Dockerfile lite, docker login

* fix(test): remove dupe flag on unwind tests

---------

Co-authored-by: Valentin Staykov <79150443+V-Staykov@users.noreply.github.com>

tclemos committed:
add stage sequence execute unit test (#1536)

V-Staykov committed:
feat: add debug_traceBatchByNumber (#1537)

* feat: add debug_traceBatchByNumber

* chore: generate endpoints doc

MorettiGeorgiev committed:
feat: add flag for empty block close time (#1542)

* feat: add flag for empty block close time

* fix: unit test stage sequence execute

* fix: use builtBlockElements.transactions instead of transactionHashesToSlots

V-Staykov committed:
feat(tx_pool): check if trace log level is set before using trace log (#1545)

* feat(tx_pool): check if trace log level is set before using trace log

* fix: tests

V-Staykov committed:
perf: remove unneded mem alloc in combine collectors (#1546)

V-Staykov committed:
fix: TestSpawnSequencingStage batch close time (#1549)

hexoscott committed:
recurring gas price check from the L1 (#1543)

Co-authored-by: Valentin Staykov <79150443+V-Staykov@users.noreply.github.com>

V-Staykov committed:
fix: remove logger setup doubling (#1550)

* fix: remove logger setup doubling

* fix: remove commented stuff

iszubok committed:
DS client - add TLS support (#1553)

Add zkevm.l2-datastreamer-use-tls flag
If set to true, a TLS connection will be used.
This also enables SNI (Server Name Indication) support and allows connecting to ds server behind virtual hosting/load balancer

V-Staykov committed:
fix: datastream client test timeouts (#1552)

MorettiGeorgiev committed:
fix: http and ws server to work on the same port

MorettiGeorgiev committed:
chore: remove comments

MorettiGeorgiev committed:
refactor: add createApiHandler() function

MorettiGeorgiev committed:
Merge pull request #1565 from 0xPolygonHermez/fix/ws-http-same-port

Fix/ws http same port

revitteth committed:
tweak(witness): stateless vm config (#1567)

Stefan-Ethernal committed:
L1 contracts event hashes unit test (#1561)

* test: implement l1 contracts event hash unit test

* fix: use modulo for indexing ethermans and remove redundant return statement

V-Staykov committed:
fix: verbosity flag set (#1556)

revitteth committed:
feat(seq): ignore bad batches check flag (#1576)

V-Staykov committed:
fix: no resequence if l1 sync enabled (#1578)

V-Staykov committed:
feat: panic if sequencer batch progress behind last sequenced batch аnd not l1 recovery mode (#1581)

* feat: panic if sequencer batch progress behind last sequenced batch and not l1 recovery mode

* fix: nil check

revitteth committed:
fix(seq): purge mined txs after mdbx tx commit (#1579)

xavier-romero committed:
devtooling tests (#1551)

jhkimqd committed:
fix: add case for sequenceBatchesValidiumBanana (#1575)

Signed-off-by: Ji Hwan <jkim@polygon.technology>

louisliu2048 committed:
cache tx sender to reduce calculate it again (#1062)

* cache tx sender to reduce calculate it again

* use map to replace sync.map

---------

Co-authored-by: Valentin Staykov <79150443+V-Staykov@users.noreply.github.com>

V-Staykov committed:
feat: add panic and readme on l1 recovery on pre forkid8 (#1580)

revitteth committed:
tweak(counters): poseidon 30 to 31 (#1587)

MorettiGeorgiev committed:
feat: add zkevm.crash-on-reorg flag

revitteth committed:
tweak(config): bali example l2 rpc url (#1584)

MorettiGeorgiev committed:
Merge branch 'zkevm' into feat/crash-on-reorg

revitteth committed:
tweak(test): send tx test reliability tweak (#1586)

MorettiGeorgiev committed:
feat: change zkevm.crash-on-reorg to zkevm.panic-on-reorg

MorettiGeorgiev committed:
Merge branch 'feat/crash-on-reorg' of github.com:0xPolygonHermez/cdk-erigon into feat/crash-on-reorg

MorettiGeorgiev committed:
feat: add zkevm.panic-on-reorg to README

MorettiGeorgiev committed:
Merge branch 'zkevm' into feat/crash-on-reorg

V-Staykov committed:
fix: datastream tests intermittent fail (#1591)

MorettiGeorgiev committed:
Merge pull request #1590 from 0xPolygonHermez/feat/crash-on-reorg

feat: add zkevm.crash-on-reorg flag

hexoscott committed:
allow the sequencer to always check for new fork information (#1606)

cffls committed:
Fix resequence (#1607)

* fix resequencing

* fix

* Add nonce check in resequence test

revitteth committed:
tweak(github): add codeowners file (#1610)

hexoscott committed:
force inclusion of info tree index updates into the witness (#1612)

IvanBelyakoff committed:
feat: add flag to allow local sequencer to shadow the main one (#1608)

IvanBelyakoff committed:
tweak(test): fix reliability of jsonrpc/send_transaction_test (#1616)

IvanBelyakoff committed:
configurable witness unwind limit (#1617) (#1618)

Co-authored-by: Scott Fairclough <70711990+hexoscott@users.noreply.github.com>

zjg555543 committed:
Add ds max chann entry config (#1597)

* update

* Format code

IvanBelyakoff committed:
fix: crash on gas tracker lowest price nil, if reject low price is on and (#1626)

frequency is 0

IvanBelyakoff committed:
fix: wrong mutex used on getLowestPrice in gas tracker (#1628)

IvanBelyakoff committed:
fix: crash on nil tx in when removing mined transactions (#1629)

IvanBelyakoff committed:
chore: Add shadow-sequencer flag desctription to README (#1638)

IvanBelyakoff committed:
chore: add tests running on stable* branches for push events (#1639)

elliothllm committed:
feat(witness-cache) write witness cache by batch and purge via flag

feat(witness-cache) write witness cache per verified batch and offset wip

feat(witness-cache) truncate witness cache properly

feat(witness-cache) ad purge witness cache to prestart

feat(web) removed logs for skipping witness cache stage

feat(witness-cache) only cache trimmed and only get cached trimmed on rpc call

feat(witness-cache) revert buildsmtfromwitness to take single witness

afa7789 committed:
chore(tools): Adding documentation to the folders where we have scripts/tools/hacks to help develop cdk-erigon (#1634)

* docs: update README.md with detailed usage instructions and supported actions for Hack tool

* docs: enhance README.md with new tools and usage instructions for Hack, Allocs, RPC Cache, RPC Checker, and Scripts

* docs: add new debugging tools and usage instructions to README files

* docs: add new debugging tools and usage instructions to README files

hexoscott committed:
Fix/stop repeat overflow transactions (#1645)

* Fix/stop repeat overflow transactions

* fix logging of hash in pool for counter overflows

* better handling of single transactions overflowing batches

* better handling of hash logging fix in pool for overflows

* resolve test issues around bad tx hash table

IvanBelyakoff committed:
fix: recover sender only for those txs that are included (#1636)

* fix: recover sender only for those txs that are included

* Moved recovering sender out of addTransaction.
Default context used for sender recovery.

afa7789 committed:
refactor: remove datastream version flag and update related configurations

afa7789 committed:
remove exported const of Datastream version.

elliothllm committed:
feat(sync-batches) sync to verified batch limit plus n batches configurable (#1652)

revitteth committed:
fix(stream_client): reconnect on GetHeader (#1648)

afa7789 committed:
fix(tests): remove old version that is not required anymore being sent on tests

afa7789 committed:
chore(kurtosis): removing mentions of datastream version in the code

elliothllm committed:
Merge pull request #1642 from 0xPolygonHermez/feat/witness-cache-batch

feat(witness-cache) write witness cache by batch and purge via flag

afa7789 committed:
chore(kurtosis): removing datastream version with sed in kurtosis setup

afa7789 committed:
Merge branch 'zkevm' into 12-flags-deprecate-and-remove-the-zkevmdatastream-version-3-flag

afa7789 committed:
Merge pull request #1649 from 0xPolygonHermez/12-flags-deprecate-and-remove-the-zkevmdatastream-version-3-flag

refactor: remove datastream version flag and update related configurations

C7-C5 committed:
Added security tests workflow (#1653)

IvanBelyakoff committed:
fix: add check for tx is Create if init code size exceeds the (#1658)

limit

elliothllm committed:
feat(zk-config) dynamic config unmarshal single file

elliothllm committed:
feat(zk-config) create tests for unmarshal zk config

elliothllm committed:
Merge pull request #1661 from 0xPolygonHermez/feat/zk-config

feat(zk-config) dynamic config unmarshal single file

revitteth committed:
tweak(ci): remove unused hourly env check (#1663)

hexoscott committed:
0tx block handling and moving to timers from tickers (#1670)

hexoscott committed:
sender lock to help prevent incorrect nonce checks (#1643) (#1671)

* sender lock to help prevent incorrect nonce checks (#1643)

* sender lock to help prevent incorrect nonce checks

helps to ensure the pool has finished working with a sent transaction before allowing any nonce checks to be performed.  Only applies to pending nonce, latest won't use this lock.

* tidy up on waiting for sender lock

elliothllm committed:
feat(zk-config-merge): create tool for merging dynamic config into 1 (#1672)

* feat(zk-confog-merge): create tool for merging dynamic config into 1

* feat(zk-confog-merge): adjust tool so that you only provide the location of the files

elliothllm committed:
fix(bali-test-code): remove bali test code from stage

fix(bali-test-code): remove prntln from rpc cache check

fix(bali-test-code): fix import

revitteth committed:
tweak(ci): stable branches triggers (#1675)

* tweak(ci): stable branches triggers

* tweak(ci): upgrade deprecated upload-artifact job

elliothllm committed:
Merge pull request #1676 from 0xPolygonHermez/fix/bali-test-code-remove

fix(bali-test-code): remove bali test code from stage

C7-C5 committed:
Added network for kurtosis

C7-C5 committed:
Merge branch 'zkevm' into feature/security-tests-fix

C7-C5 committed:
Merge pull request #1679 from 0xPolygonHermez/feature/security-tests-fix

feature(qa): added network into security tests

elliothllm committed:
update(readme): update readme for require ubuntu version 22 or higher

elliothllm committed:
Merge pull request #1683 from 0xPolygonHermez/update/readme-ubuntu-requirement

update(readme): update readme for require ubuntu version 22 or higher

IvanBelyakoff committed:
fix: test sequence stage timeouts (#1685)

revitteth committed:
tweak(ci): lint docs on stable (#1680)

revitteth committed:
tweak(config): add shadow-sequencer true to devnet config (#1678)

hexoscott committed:
Fix/2.60 upgrade fork changes (#1631) (#1687)

* sequencer to go through stage loop whilst halted

* updates to L1 processing when network is in a halted state
# Conflicts:
#	zk/stages/stage_batches.go

elliothllm committed:
feat(bad-tx): trim bad tx hashes from db via config

feat(bad-tx): rm println

elliothllm committed:
feat(bad-tx): add unit test for truncate bad txs

feat(bad-tx): unwind test fix

feat(verify-flag): unwind test pahse 2 fix

elliothllm committed:
feat(zk-stages): add metrics for zy sync and version

feat(zk-stages): imports

feat(zk-stages): imports

hexoscott committed:
Stage interhashes optimisations (#1655)

* first pass switching SMT algo to use ETL

moving ETL code behind the smt DB interfaces

WIP - adding timing logging

first pass of inters stage improvements

profiling added to inters stage

remove log timing from inters sub processes

more efficient node key to hex conversion

moving some small functions away from big int usage

* switch etl to sortable buffer for inters

elliothllm committed:
feat(zk-stages): zk stages metrics

feat(zk-stages): whitespace

elliothllm committed:
feat(zk-stages): rm xstring deps from eth backend

elliothllm committed:
feat(verify-flag): verify datastreamer flag on startup

elliothllm committed:
feat(verify-flag): only do the verify if port is not abcent

elliothllm committed:
Merge pull request #1691 from 0xPolygonHermez/feat/verify-ds-flag-url

feat(verify-flag): verify datastreamer flag on startup

elliothllm committed:
feat(verify-flag): avoid mdbx overlapping transaction same thread

elliothllm committed:
Merge pull request #1690 from 0xPolygonHermez/feat/zk-stages-metrics

feat(zk-stages): add metrics for zk sync and version

elliothllm committed:
Merge pull request #1688 from 0xPolygonHermez/feat/trim-badtx-hashes

feat(bad-tx): trim bad tx hashes from db via config

hexoscott committed:
Inters speed up 2 (#1694)

* moving from big int usage for storage key calculation

* use common package to avoid big int when parsing an address

* alternative scalar to array using uint64 rather than big int

* remove unused util function

hexoscott committed:
Inters speed up 3 (#1699)

* move nodevalue8 to uint64

* node value 12 from big int to uint64

* memdb changes to use uint64 rather than big int for keys

praetoriansentry committed:
fix: switching to thread 0

Hoping to address
https://github.com/0xPolygonHermez/cdk-erigon/issues/1697 by
defaulting to thread 0 rather than 1

revitteth committed:
fix(counters): smt depth fix and test (#1695)

* fix(counters): smt depth fix and test

* fix(smt): set depth and get depth correctly

* fix(smt): depth update

praetoriansentry committed:
Merge branch 'zkevm' into jhilliard/secp2561k1-cpu-count-fix

praetoriansentry committed:
Merge pull request #1701 from 0xPolygonHermez/jhilliard/secp2561k1-cpu-count-fix

Switching from thread 1 to 0 for `secp256k1.ContextForThread`

IvanBelyakoff committed:
fix: nil crash on uninitialized block var (#1705)

elliothllm committed:
fix(tx-gas-logger): recover from panic on logger (#1702)

* fix(tx-gas-logger): recover from panic on logger if tx becomes nil during metrics collection

* fix(tx-gas-logger): add select and check form ctx done and log recover

* tweak(mdbx): kv_mdbx add early return in CollectMetrics on nil receiver

---------

Co-authored-by: Max Revitt <max@revitt.consulting>

elliothllm committed:
fix(eth-logs): create eth logs max limit flag (#1706)

* create eth logs max limit flag
fix(eth-logs): create eth logs max limit flag to prevent ddos risk

fix(eth-logs): fix tests

* fix(eth-logs): change message to eth logs max range message to block range too large

* fix(eth-logs): add unit test for eth get logs using log max range

* tweak(api): add block range max to limit response

---------

Co-authored-by: Max Revitt <max@revitt.consulting>

IvanBelyakoff committed:
fix: 0 tx blocks after limbo (#1708)

* fix: 0 tx blocks after limbo

* Add executor-enabled flag.

It disables using remote executor even when executor-urls option is filled in,
which is helpful when testing limbo mode, when txs are still sent to
the remote executor.

IvanBelyakoff committed:
fix: crash on nil block if no txs mined (#1715)

elliothllm committed:
fix(gas-pool): create a gas pool per block in normalcy mode on sequencer (#1712)

fix(gas-pool): changed name normalcy gas pool

elliothllm committed:
fix(gas-pool): fix for naming on gas pool and pull to block level in sequencer (#1723)

elliothllm committed:
feat(health-check): add tx pool and last block mined to health check (#1724)

* feat(health-check): add tx pool and last block mined to health check

feat(health): health wip 2

feat(health): add headers

feat(health): add tx pool and las block to tests

* feat(health): check last 5 blocks for 0tx blocks on pendiong tc pool

feat(health): check for transactions found within first loop

hexoscott committed:
Zkevm backport from 2.61 (#1726)

* commit block update to state before removing mined transactions

without this change there is a very small window of time where you
could request a pending nonce and the pool has had the transactions
removed before the latest executed block is updated in the DB so you
in fact get the "latest" nonce from state for an old block instead.

we also slightly reduce the locks on the pool by combining the two slices
of transactions that we want to remove from the pool.

* deadlocks

* pool to manage flush locking internally

* acquire semaphore before tracking tx begin

* set no gossip to default to true

This really isn't needed in zkevm networks as we don't support p2p gossip
and it creates load on the DB that we simply don't need.

Some of the deadlocks we've seen, once released, log errors from the
gossip code on repeat so better we don't attempt it at all in this case

Setting the flag to false will override this default so we can re-enable
when the time suits

* do not lock on waiting for the open transactions to close

if we lock here then we can't close down already open transactions
because they will deadlock as they enter the function

* update to mdbx deadlock code test

allows the test to run to completion without thread exhaustion

* useful comment about the stacked locks in the pool best function

* enable pool p2p gossip for unit tests that rely on the old default behaviour

* remove unnecessary 2nd RO tx in send transaction

* do not call track end on error acquiring the semaphore

* increase timeout in stream client tests

* reduce logging noise for common complaints (#1722)

* temporarily halt processing tx for a sender when they encounter overflow

* temporarily halt processing tx for a sender when they encounter overflow

previously during an overflow situation we would continue to attempt to process
transactions for the same sender that were yielded.  In networks that have a few
very busy accounts this created nonce too high problems which we then incorrectly
attempted to remove from the pool.  Eventually this led to transactions sitting in
the queued pool that actually had a valid nonce and should be processed.

Now we pause processing for a sender that encountered an overflow until the next batch
so that we don't create these problems.

* adjustments to sendersToSkip after changes to sender recovery logic

* adjustment of error handling

no more special treatment for nonce too high.  They stick around in
the pool for too long and we keep attempting to process them so they
should be discarded like the others.

this will also update other transactions in the pool for the same
sender and move those over into the queued pool which also helps

* remove mined transactions per block rather than per batch

revitteth committed:
tweak(deps): upgrade datastreamer (#1718)

elliothllm committed:
feat(debug-tools): made tool for comparing mdbx dbs (#1729)

leovct committed:
fix: secp256k1 thread (#1728)

hexoscott committed:
cleaner shut down (#1734)

shared mutex in the sync.Cond caused a panic on unlocking an unlocked
 mutex as it was shared with other logic

elliothllm committed:
tweak(tx-pool-grpc): add sub pool type to panic (#1731)

tweak(tx-pool-grpc): log error but still if not valid type

tweak(tx-pool-grpc): add sender

hexoscott committed:
do not commit new block details to the DB during limbo processing (#1738)

revitteth committed:
improve(acl): ignore dbinfo in unwind tests (#1744)

Co-authored-by: Elliot Hallam <elliot@revitt.consulting>

zjg555543 committed:
Batch commit zkHash data for sequencer  (#1730)

* update

* update

hexoscott committed:
nonce issue detection and remedy in execution (#1746)

* nonce issue detection and remedy in execution

here we pause handling further transactions for the sender in this
batch and also trigger the pool to perform a sender state change to
remove nonce too low transactions and move nonce too high to queued

* mop up discarded transactions during yielding correctly

* Ignore DbInfo in unwind test

zjg555543 committed:
Optimize calculating of the L1 Tree (#1747)

* update

* update

IvanBelyakoff committed:
fix: respect transaction execution counters (#1749)

IvanBelyakoff committed:
fix: unreadable symbols in DS stream on client side (#1698)

* fix: unreadable symbols in DS stream on client side

* fix: improved handling DS stop command and streaming status

KyrinCode committed:
fix empty forkid and parenthash in l2block read from datastream

elliothllm committed:
tweak(witness-cache): tweak witness cache flags for offset ahead and below

elliothllm committed:
Merge pull request #1764 from 0xPolygonHermez/tweak/witness-cache-offset

tweak(witness-cache): witness cache flags offset ahead and below

elliothllm committed:
tweak(witness-cache): cache 1 down from highest seen

elliothllm committed:
fix(rpc): stop getblockhash from crashing when sending hash (#1761)

fix(rpc): allow block hash for eth get receipts

hexoscott committed:
lock the pool flush mutex before calling yield (#1759)

There was a race condition here where opening the db view for yielding
didn't contain info from the DB that wasn't flushed completely yet.
Now the flush can't start running until the DB view has been used
and vice versa for yielding, we now wait for the flush to finish.

Tested using the polycli load tool which is what highlighted this issue
in the first place

elliothllm committed:
Merge pull request #1769 from 0xPolygonHermez/tweak/witness-cache-limit

tweak(witness-cache): cache 1 down from highest seen

elliothllm committed:
feat(info-tree-compare): script to check info trees on 2 nodes (#1753)

elliothllm committed:
tweak(unwind): remove pipefall from unwind script (#1773)

dloghin committed:
improve(smt/pkg/utils): improve the performance of HashContractBytecodeBigInt() (#1772)

* improve the performance of HashContractBytecodeBigInt()

* update go.sum

* update poseidongold to v0.0.3

revitteth committed:
Stable 2.61 merge to zkevm (#1783)

* Merge pull request #1565 from 0xPolygonHermez/fix/ws-http-same-port

Fix/ws http same port

* tweak(counters): poseidon 30 to 31 (#1587) (#1589)

* add flag to reject transactions below default config (#1602) (#1604)

* 2.61 fork upgrade changes (#1615)

* allow the sequencer to always check for new fork information (#1606)

* include all used info tree indexes in the witness

* configurable witness unwind limit (#1617)

* Fix/2.61 upgrade fork changes (#1630)

* sequencer to go through stage loop whilst halted

* updates to L1 processing when network is in a halted state

* Fix/stop repeat overflow transactions (#1640) (#1644)

* fix logging of hash in pool for counter overflows

* better handling of single transactions overflowing batches

* better handling of hash logging fix in pool for overflows
 Conflicts:
	zk/hermez_db/db.go
	zk/stages/stage_sequence_execute.go

* resolve test issues around bad tx hash table (#1646)

* 0tx block handling and moving to timers from tickers (#1673)

* 0tx block handling and moving to timers from tickers

* tweak(ci): stable branches triggers (#1675)

* run ci on stable branches

* tweak(ci): lint docs on stable

---------

Co-authored-by: Max Revitt <max.revitt@gmail.com>
Co-authored-by: Max Revitt <max@revitt.consulting>

* sender lock to help prevent incorrect nonce checks (#1643) (#1671) (#1681)

* sender lock to help prevent incorrect nonce checks (#1643)

* sender lock to help prevent incorrect nonce checks

helps to ensure the pool has finished working with a sent transaction before allowing any nonce checks to be performed.  Only applies to pending nonce, latest won't use this lock.

* tidy up on waiting for sender lock

* DS client - add TLS support (#1553) (#1682)

Add zkevm.l2-datastreamer-use-tls flag
If set to true, a TLS connection will be used.
This also enables SNI (Server Name Indication) support and allows connecting to ds server behind virtual hosting/load balancer

Co-authored-by: Ivan Zubok <93441191+iszubok@users.noreply.github.com>

* fix: test SpawnSequencingStage timeouts issue (#1684)

* perf: remove unneded mem alloc in combine collectors (#1546) (#1547) (#1686)

Co-authored-by: Valentin Staykov <79150443+V-Staykov@users.noreply.github.com>

* fix: add check for tx is Create if init code size exceeds the (#1659)

limit

Co-authored-by: Max Revitt <max.revitt@gmail.com>

* fix: switching to thread 0 (#1707)

* fix: 0 tx blocks after limbo (#1708) (#1711)

* fix: 0 tx blocks after limbo

* Add executor-enabled flag.

It disables using remote executor even when executor-urls option is filled in,
which is helpful when testing limbo mode, when txs are still sent to
the remote executor.

* fix: recover sender only for those txs that are included (#1636) (#1710)

* fix: recover sender only for those txs that are included

* Moved recovering sender out of addTransaction.
Default context used for sender recovery.

* commit block update to state before removing mined transactions (#1717)

without this change there is a very small window of time where you
could request a pending nonce and the pool has had the transactions
removed before the latest executed block is updated in the DB so you
in fact get the "latest" nonce from state for an old block instead.

we also slightly reduce the locks on the pool by combining the two slices
of transactions that we want to remove from the pool.

* fix: crash on nil block if no txs mined (#1716)

* 2.61 deadlocks (#1713)

* pool to manage flush locking internally

* acquire semaphore before tracking tx begin

* set no gossip to default to true

This really isn't needed in zkevm networks as we don't support p2p gossip
and it creates load on the DB that we simply don't need.

Some of the deadlocks we've seen, once released, log errors from the
gossip code on repeat so better we don't attempt it at all in this case

Setting the flag to false will override this default so we can re-enable
when the time suits

* do not lock on waiting for the open transactions to close

if we lock here then we can't close down already open transactions
because they will deadlock as they enter the function

* update to mdbx deadlock code test

allows the test to run to completion without thread exhaustion

* useful comment about the stacked locks in the pool best function

* enable pool p2p gossip for unit tests that rely on the old default behaviour

* remove unnecessary 2nd RO tx in send transaction

* do not call track end on error acquiring the semaphore

* increase timeout in stream client tests

* reduce logging noise for common complaints (#1722)

* temporarily halt processing tx for a sender when they encounter overflow (#1709)

* temporarily halt processing tx for a sender when they encounter overflow

previously during an overflow situation we would continue to attempt to process
transactions for the same sender that were yielded.  In networks that have a few
very busy accounts this created nonce too high problems which we then incorrectly
attempted to remove from the pool.  Eventually this led to transactions sitting in
the queued pool that actually had a valid nonce and should be processed.

Now we pause processing for a sender that encountered an overflow until the next batch
so that we don't create these problems.

* adjustments to sendersToSkip after changes to sender recovery logic

* adjustment of error handling (#1725)

no more special treatment for nonce too high.  They stick around in
the pool for too long and we keep attempting to process them so they
should be discarded like the others.

this will also update other transactions in the pool for the same
sender and move those over into the queued pool which also helps

* fix(gas-pool): create a gas pool per block in normalcy mode on sequencer [stable 2.61] (#1720)

* fix(gas-pool): create a gas pool per block in normalcy mode on sequencer

fix(gas-pool): changed name normalcy gas pool

* fix(gas-pool): fix for naming on gas pool and pull to block level in sequencer

---------

Co-authored-by: Elliot Hallam <elliot@revitt.consulting>

* tweak(deps): upgrade datastreamer (#1719)

* fix(eth-logs): create eth logs max limit flag (#1706) (#1735)

* create eth logs max limit flag
fix(eth-logs): create eth logs max limit flag to prevent ddos risk

fix(eth-logs): fix tests

* fix(eth-logs): change message to eth logs max range message to block range too large

* fix(eth-logs): add unit test for eth get logs using log max range

* tweak(api): add block range max to limit response

---------

Co-authored-by: Max Revitt <max@revitt.consulting>

tweak(eth-logs): tests

* tweak(tx-pool-grpc): add sub pool type to panic (#1733)

tweak(tx-pool-grpc): log error but still if not valid type

tweak(tx-pool-grpc): add sender

* cleaner shut down (#1734) (#1736)

shared mutex in the sync.Cond caused a panic on unlocking an unlocked
 mutex as it was shared with other logic

* do not commit new block details to the DB during limbo processing (#1737)

* do not commit new block details to the DB during limbo processing

* do not clean up on unwind test completion

* nonce issue detection and remedy in execution (#1741)

* nonce issue detection and remedy in execution

here we pause handling further transactions for the sender in this
batch and also trigger the pool to perform a sender state change to
remove nonce too low transactions and move nonce too high to queued

* mop up discarded transactions during yielding correctly

* Ignore DbInfo in unwind test

* fix: respect transaction execution counters (#1757)

* lock the pool flush mutex before calling yield (#1758)

There was a race condition here where opening the db view for yielding
didn't contain info from the DB that wasn't flushed completely yet.
Now the flush can't start running until the DB view has been used
and vice versa for yielding, we now wait for the flush to finish.

Tested using the polycli load tool which is what highlighted this issue
in the first place

* fix(counters): smt depth fix and test (#1695) (#1751)

* fix(counters): smt depth fix and test

* fix(smt): set depth and get depth correctly

* fix(smt): depth update

Co-authored-by: Max Revitt <max.revitt@gmail.com>

* fix: nil crash on uninitialized block var (#1763)

* fix: unreadable symbols in DS stream on client side (#1750)

* feat(witness-cache) write witness cache by batch and purge via flag

feat(witness-cache) write witness cache per verified batch and offset wip

feat(witness-cache) truncate witness cache properly

feat(witness-cache) ad purge witness cache to prestart

feat(web) removed logs for skipping witness cache stage

feat(witness-cache) only cache trimmed and only get cached trimmed on rpc call

feat(witness-cache) revert buildsmtfromwitness to take single witness

* tweak(witness-cache): tweak witness cache flags for offset ahead and below

* tweak(witness-cache): cache 1 down from highest seen

* fix(rpc): stop getblockhash from crashing when sending hash (#1761)

fix(rpc): allow block hash for eth get receipts

* improve(smt/pkg/utils): improve the performance of HashContractBytecodeBigInt() (#1772) (#1780)

* improve the performance of HashContractBytecodeBigInt()

* update go.sum

* update poseidongold to v0.0.3

Co-authored-by: Dumi Loghin <dumiloghin@gmail.com>

* fix(merge): fix merge artifact

* fix(merge): dupe flag

* fix(execute): separate flush context on shutdown

---------

Co-authored-by: Moretti Georgiev <moretti.georgiev@gmail.com>
Co-authored-by: Scott Fairclough <70711990+hexoscott@users.noreply.github.com>
Co-authored-by: Ivan Zubok <93441191+iszubok@users.noreply.github.com>
Co-authored-by: Ivan Belyakov <ivan.belyakov.job@gmail.com>
Co-authored-by: Valentin Staykov <79150443+V-Staykov@users.noreply.github.com>
Co-authored-by: Elliot Hallam <elliot@revitt.consulting>
Co-authored-by: elliothllm <143621200+elliothllm@users.noreply.github.com>
Co-authored-by: Dumi Loghin <dumiloghin@gmail.com>

dloghin committed:
implement LRU cache for decoded tx and remove duplicated sleep on SequencerTimeoutOnEmptyTxPool (#1775)

elliothllm committed:
improve(acl): allow for a json acl to be unmarshalled and updated. (#1742)

* improve(acl): allow for a json acl to be unmarshalled and updated in the db

improve(acl): only update with flag

improve(acl): update improvements

improve(acl): update improvements

* improve(acl): ignore dbinfo in unwind tests

* improvement(acl): use in memory acl list without mdbx

* improvement(acl): multiple test cases for acl rules

elliothllm committed:
enhance(l1-info-sync): sync l1 info table from l2 rpc id an rpc node (#1696)

* enhance(l1-info-sync): sync l1 info table from l2 rpc id an rpc node

enhance(l1-info-sync): l2 syncer wip

enhance(l1-info-sync): l2 syncer

enhance(l1-info-sync): fix tests

* feat(info tree): changed name around the l2 rpc info tree syncer

hexoscott committed:
Zkevm sequencer as rpc (#1732)

* trigger notifications per block on the sequencer

* fixing up unit tests relating to notification handling

* adjust timings on sequencer test to help CI

revitteth commented:
@KyrinCode please could you try to bump these PRs to get the cla-bot to respond? I can't manually re-run it seems.

revitteth commented:
With regards to this PR in particular - I'm going to come back to it later - need some time to check some things with it. So for now we will park if that's ok.

elliothllm committed:
feat(tx-pool-metrics): calculate txs in out and median wait time (#1752)

* feat(tx-pool-metrics): calculate txs in out and median wait time

tweak(tx-pool): add metrics

tweak(tx-pool): metrics

tweak(tx-pool): metrics run time

* tweak(tx-pool): pre allocate memory for wait times

tweak(tx-pool): pre allocate memory for wait times

* feat(txpool-metrics): add metrics to prometheus

feat(txpool-metrics): remove sort as they will come sorted

revitteth committed:
[Poseidon] - Fix  (#1787)

* Revert "improve(smt/pkg/utils): improve the performance of HashContractBytecodeBigInt() (#1772)"

This reverts commit 607ab1389f66fc2fc95ac2cc15fe41b421137fb4.

* fix(poseidon): hashing algorithm only

revitteth committed:
fix(debug): ensure err returned for debug flag so the node exits (#1785)

* fix(debug): ensure err returned for debug flag so the node exits

* fix(debug): no error on stopped

elliothllm committed:
feat(debug-flag): add debug flag to disable state root check (#1791)

feat(debug-flag): add config to http cfg

feat(debug-flag): add config to tests

elliothllm committed:
tweak(l1-syncer): panic on exceeding 20 retries - [Requires Alert] (#1790)

* tweak(l1-syncer): panic on exceeding 20 retries

* tweak(l1-syncer): add retry num to error

elliothllm committed:
fix(witness-cache): calculate start and from batch with offsets (#1793)

IvanBelyakoff committed:
feat: check counters overflow on eth_estimateGas (#1795)

IvanBelyakoff committed:
Fix/estimateGas ignores counters if disable-virtual-counters=true (#1799) (#1800)

IvanBelyakoff committed:
chore: change some datastreamer logs to debug level (#1801)

KyrinCode committed:
Merge branch 'zkevm' into kyrin/fix-empty-forkid-parenthash-of-l2block-from-datastream
